src/monstergenerator.cpp

```diff
@@ -888,7 +888,7 @@ void MonsterGenerator::load_species( const JsonObject &jo, const std::string &sr
 void species_type::load( const JsonObject &jo, const std::string & )
 {
     optional( jo, was_loaded, "description", description );
-    optional( jo, was_loaded, "footsteps", footsteps, "footsteps." );
+    optional( jo, was_loaded, "footsteps", footsteps, _( "footsteps." ) );
```

Comment:
```suggestion
    optional( jo, was_loaded, "footsteps", footsteps, translate_marker( "footsteps." ) );
```
So it doesn't translate the string twice at line 892.

Also, as the code currently stands, the translation won't update when switching the language in game. Could you change the type of `footsteps` to `translation` and update the code accordingly to fix it and modernize the code?

Reply:
Thanks for your review. 

> Could you change the type of `footsteps` to `translation`

Could you provide a simple example? `footsteps` is defined in the following structure:

```c++
struct species_type {
    species_id id;
    bool was_loaded = false;
    translation description;
    std::string footsteps;
    enum_bitset<m_flag> flags;
    enum_bitset<mon_trigger> anger;
    enum_bitset<mon_trigger> fear;
    enum_bitset<mon_trigger> placate;
    std::string get_footsteps() const {
        return footsteps;
    }

    species_type(): id( species_id::NULL_ID() ) {

    }

    void load( const JsonObject &jo, const std::string &src );
};
```

Reply:
Change `std::string footsteps;` to `translation footsteps;`, and update the code where it is used. Use `to_translation( "footsteps." )` to create the default value, and `footstep.translated()` to get the translated string.

Reply:
I changed the type of variable. I'm not sure that everything is correct, please review it. In the game, this solution works, I tested.

Reply:
> Also, as the code currently stands, the translation won't update when switching the language in game.

There were some performance considerations to do so when it was jsonized (see https://github.com/CleverRaven/Cataclysm-DDA/pull/31493)

Reply:
@ZhilkinSerg Should I return the type of the variable as it was before?

Reply:
We can investigate the performance problem anytime, but for now we support switching language without restart, so this ensures consistency. Also I don't think this particular string is used that frequently.

Anyway, here are two (immature) ideas of mine about performance optimization of the translation class: we can create a new class `cached_translation`, which stores an extra cached translated string and the language of the cached string (perhaps through an id or a reference), and in the next translation call, we check if the language has changed and update the cache accordingly. This of course requires a few more bytes per translation object for the cache.

Or we can add raw strings and their translation to a central repository when creating `translation` objects, which then points to the translated strings through a reference. When switching the language, we iterate through the central repo and update the translated string with gettext calls, so the translation objects automatically point to the new translated strings. I think this one is superior since it not only addresses the performance issue, but also reduces the memory usage of the translation class.